vowpalwabbit/example_predict.h

```diff
@@ -19,36 +20,33 @@ struct example_predict
     namespace_index* _index;
 
    public:
-    iterator(features* feature_space, namespace_index* index) : _feature_space(feature_space), _index(index) {}
-
-    features& operator*() { return _feature_space[*_index]; }
-
-    iterator& operator++()
-    {
-      _index++;
-      return *this;
-    }
-
-    namespace_index index() { return *_index; }
-
-    bool operator==(const iterator& rhs) { return _index == rhs._index; }
-    bool operator!=(const iterator& rhs) { return _index != rhs._index; }
+    iterator(features* feature_space, namespace_index* index);
```

Comment:
nit: maybe add a quick comment that modifying the indices can invalidate these iterators